ui/raidboss/data/03-hw/trial/ravana-ex.txt

```diff
@@ -1,107 +1,174 @@
+290.9 "Atma-Linga" sync /:Ravana:EA6:/
+296.2 "Tapasya x3" sync /:Ravana:EA4:/
+303.3 "Atma-Linga" sync /:Ravana:EA6:/
+306.4 "Blinding Blade" sync /:Ravana:E9F:/
+309.5 "Atma-Linga" sync /:Ravana:EA6:/
+312.7 "Tapasya x3" sync /:Ravana:EA4:/
+320.8 "Blinding Blade" sync /:Ravana:E9F:/
+325.5 "The Seeing" sync /:Ravana:EA[012]:/
+330.7 "Atma-Linga x2" # sync /:Ravana:EA6:/
+
+# The intermission can have 0-10 Laughing Moon charges before Ravana using the Chandrahas ultimate.
+# The number is dependent on how many Gana are able to finish their Falling Laughter casts.
+# More than 1-2 is typically a wipe, as each inflicts a Vulnerability Up stack to the party.
+341.1 "Bloody Fuller" sync /:Ravana:EB3:/ window 341.1,5
+341.1 "--untargetable--"
+
+400.0 "Chandrahas" sync /:Ravana:EB5:/ window 400,5
+413.9 "Beetle Avatar" sync /:Ravana:E82:/ window 15,15
+414.0 "--targetable--" # Technically slightly before, but invulnerability drops here.
+422.0 "Pillars Of Heaven" sync /:Ravana:EB8:/
+429.1 "--sync--" sync /:Ravana:EB9:/
+429.8 "Laughing Rose" # The EB9 skill resolves well before the damage displays
+445.4 "Surpanakha x4" # sync /:Ravana:EBA:/
+453.1 "The Rose Of Conviction" sync /:Ravana:EB6:/
+460.2 "The Rose Of Hate" sync /:Ravana:EBB:/
+465.4 "Surpanakha x7" # sync /:Ravana:EBA:/
+466.3 "The Rose Of Conquest" sync /:Ravana's Will:EB7:/
+485.8 "Pillars Of Heaven" sync /:Ravana:EB8:/
+493.0 "--sync--" sync /:Ravana:EB9:/
+493.7 "Laughing Rose"
+506.2 "Surpanakha x7?" # sync /:Ravana:EBA:/
+# This phase ends at 100 Bloodlust.
+
+514.0 "Scorpion Avatar" sync /:Ravana:E81:/ window 114,5
+514.1 "--sync--" sync /:Ravana:13BF:/
+519.2 "Blades Of Carnage And Liberation" sync /:Ravana:137A:/
+540.4 "Swift Liberation" sync /:Ravana:EA7:/
+540.4 "--untargetable--"
+540.5 "Swift Liberation Dash 1" # sync /:Ravana:EA8:/
+543.5 "Swift Liberation Dash 2" # sync /:Ravana:EA8:/
+546.5 "Swift Liberation Dash 3" # sync /:Ravana:EA8:/
+549.5 "Swift Liberation Dash 4" # sync /:Ravana:EA8:/
+552.7 "Swift Liberation (Flames)" # sync /:Ravana:EA9:/
+556.9 "Swift Liberation (Circles)" # sync /:Ravana:EAB:/
+562.1 "--targetable--"
+562.3 "Blades Of Carnage And Liberation" sync /:Ravana:137A:/ window 20,20
+583.5 "Final Liberation" sync /:Ravana:EAC:/
+583.5 "--untargetable--"
+583.6 "Final Liberation (1st Double Prey)" sync /:Ravana:EAD:/
+591.6 "Final Liberation (2nd Double Prey)" sync /:Ravana:EAD:/
+597.7 "--sync--" sync /:Ravana:13C4:/
+597.7 "Final Liberation (Outer AoE)" sync /:Ravana:EAE:/
+598.9 "--targetable--"
+603.9 "Final Liberation (Inner AoE)" sync /:Ravana:EAF:/
+603.9 "--untargetable--"
+610.1 "--targetable--"
+
+610.2 "Dragonfly Avatar" sync /:Ravana:E80:/ window 96,20
+610.3 "--sync--" sync /:Ravana:1299:/
+617.3 "Warlord Shell" sync /:Ravana:EB1:/
+622.5 "The Seeing" sync /:Ravana:EA[012]:/
+626.7 "Atma-Linga" sync /:Ravana:EA6:/
+631.0 "The Seeing" sync /:Ravana:EA[012]:/
+640.2 "Warlord Flame" sync /:Ravana:EB2:/
```

Comment:
Yes, it should be a question mark, thanks for the catch! This is the same as in the opening Dragonfly block. Blinding Blade moves up if Warlord Flame isn't used. I've added a `startsCasting` sync here as well to make it slightly less jarring when it jumps.

Reply:
Ok, I know this is into the land of tiny details that don't really matter, but if it does adjust the time, I'm curious what you think we should do in general for abilities that won't go off in the general case.  When I did a5s, there's some extra abilities that happen if you don't kill the two fausts before hummelfaust drops: https://github.com/quisquous/cactbot/blob/master/ui/raidboss/data/03-hw/raid/a5s.txt#L28-L33

And so the timeline text and syncs assume this won't happen, but if it does, then it'll sync backwards in time to the unexpected ability, and continue continuously (non-disjointedly?) from there.  In other words, between Panzer Vor and the first Kaltstrahl it won't jump in time unless there's the unexpected abilities that show up.

Mostly just wanted your opinion on how to handle this in the general case.  Like, is this a reasonable approach? Is it better to put the full ability list and jump? Should we treat each one individually and see how we feel?

I don't know.  In this case it was a one-off thought that I maybe overcomplicated things with.  Just curious if you think it's worth doing something like this in these cases?

Reply:
No, it's a fair question. Blinding Blade is an instant cleave, so it would be nice for future timeline triggers to notify it accurately no matter what.

For the first Flame, it's perfectly fine as it's set up, since the `EB0` ability is used alongside the point where Ravana calls in the Gana adds, so no matter what happens the timeline will sync without issues. Jumps there will be expected since that's a potential push regardless of Flame.

For the second Flame, if that's how it's handled elsewhere, it's probably best to do that here too for consistency, since it's a situation where timeline triggers could be affected. I'll re-run things later today and get exact timings.

That all being said, it's a good thing you called that out, since it led me to find that Beetle doesn't automatically follow Dragonfly there. (???) [One particular report on FFLogs](https://www.fflogs.com/reports/kvPbqZXchD7njwTV#fight=3&type=damage-done) has Scorpion following instead. Somehow. I had used it before in making a prototype timeline, but only in aggregate with other timelines, and unfortunately the result was a garbled mess that I tossed out, so I had ignored this fight thereafter. I'll re-time things later today when I have a little more time and see whether that's random or definite.

WHY, Square?